pkg/config/visitor.go

```diff
@@ -196,26 +165,40 @@ func (cfg *STCPVisitorConf) Compare(cmp VisitorConf) bool {
 	if !cfg.BaseVisitorConf.compare(&cmpConf.BaseVisitorConf) {
 		return false
 	}
+
+	// Add custom login equal, if exists
+
 	return true
 }
 
-func (cfg *STCPVisitorConf) UnmarshalFromIni(prefix string, name string, section ini.Section) (err error) {
-	if err = cfg.BaseVisitorConf.UnmarshalFromIni(prefix, name, section); err != nil {
-		return
+func (cfg *STCPVisitorConf) UnmarshalFromIni(prefix string, name string, section *ini.Section) error {
+	err := section.MapTo(cfg)
+	if err != nil {
+		return err
 	}
-	return
+
+	err = cfg.BaseVisitorConf.decorate(prefix, name, section)
+	if err != nil {
+		return err
+	}
+
+	// Add custom logic unmarshal, if exists
+
+	return nil
 }
 
 func (cfg *STCPVisitorConf) Check() (err error) {
 	if err = cfg.BaseVisitorConf.check(); err != nil {
 		return
 	}
+
+	// Add custom logic validate, if exists
+
 	return
 }
 
-type XTCPVisitorConf struct {
-	BaseVisitorConf
-}
+// XTCP
+var _ VisitorConf = &XTCPVisitorConf{}
```

Comment:
Can we check if it satisfies `VisitorConf` interface in `init` functions using `VisitorConfTypeMap`?

So we don't need to write repeated content for each struct.

Reply:
actually, it is not a check in runtime,  but compile time，see: https://stackoverflow.com/questions/13194272/what-does-an-underscore-and-interface-name-after-keyword-var-mean

Reply:
I known. But it's worth nothing just increase the codes.

Each PR will run ci and e2e tests to make sure it's alright, this mistake won't be missed.